runners/flink/src/main/java/org/apache/beam/runners/flink/LyftFlinkStreamingPortableTranslations.java

```diff
@@ -193,33 +225,63 @@ public boolean isEndOfStream(WindowedValue<byte[]> nextElement) {
   }
 
   // TODO: translation assumes byte[] values, does not support keys and headers
-  private void translateKafkaSink(
+  @VisibleForTesting
+  void translateKafkaSink(
       String id, RunnerApi.Pipeline pipeline, StreamingTranslationContext context) {
     RunnerApi.PTransform pTransform = pipeline.getComponents().getTransformsOrThrow(id);
     final String topic;
-    final Properties properties = new Properties();
+    final String bootstrapServers;
+
     ObjectMapper mapper = new ObjectMapper();
+    FlinkKafkaProducer011<WindowedValue<byte[]>> producer;
+    LyftKafkaProducerBuilder<WindowedValue<byte[]>> producerBuilder =
+        new LyftKafkaProducerBuilder<>();
     try {
       Map<String, Object> params =
           mapper.readValue(pTransform.getSpec().getPayload().toByteArray(), Map.class);
 
       Preconditions.checkNotNull(topic = (String) params.get("topic"), "'topic' needs to be set");
 
-      Map<?, ?> consumerProps = (Map) params.get("properties");
-      Preconditions.checkNotNull(consumerProps, "'properties' need to be set");
-      properties.putAll(consumerProps);
+      Map<?, ?> producerProps = (Map) params.get("properties");
+      Preconditions.checkNotNull(producerProps, "'properties' need to be set");
+      logger.info(
+          "Configuring kafka producer for topic {} with properties {}", topic, producerProps);
+
+      Preconditions.checkNotNull(
+          bootstrapServers = (String) producerProps.remove("bootstrap.servers"),
+          "'bootstrap.servers' needs to be set");
+
+      final String splits[] = bootstrapServers.split(":", 2);
+      final String bootstrapServer = splits[0];
+      int port = splits.length == 2 ? Integer.parseInt(splits[1]) : DEFAULT_KAFKA_BROKER_PORT;
+      final String userName = (String) params.get("username");
+      final String password = (String) params.get("password");
+
+      if (userName != null) {
+        producerBuilder.withUsername(userName);
+      }
+
+      if (password != null) {
+        producerBuilder.withPassword(password);
+      }
+
+      for (Map.Entry<?, ?> entry : producerProps.entrySet()) {
+        producerBuilder.withKafkaProperty(
+            String.valueOf(entry.getKey()), String.valueOf(entry.getValue()));
+      }
+
+      producer =
+          producerBuilder.build(
+              topic, bootstrapServer, port, new ByteArrayWindowedValueSerializer());
+
     } catch (IOException e) {
       throw new RuntimeException("Could not parse KafkaConsumer properties.", e);
     }
 
-    logger.info("Kafka producer for topic {} with properties {}", topic, properties);
```

Comment:
Does this logging occur elsewhere or why is it removed?

Reply:
Yes! https://github.com/lyft/beam/blob/5db789871605c0406bce6a881b520fe37cdc9514/runners/flink/src/main/java/org/apache/beam/runners/flink/LyftFlinkStreamingPortableTranslations.java#L247